src/libpmemobj/sync.c

```diff
@@ -73,6 +73,10 @@ _get_lock(uint64_t pop_runid, volatile uint64_t *runid, void *lock,
 	LOG(15, "pop_runid %ju runid %ju lock %p init_lock %p", pop_runid,
 		*runid, lock, init_lock);
 
+	COMPILE_ERROR_ON(alignof(PMEMmutex) != alignof(pthread_mutex_t));
```

Comment:
I know some of you don't like C11 for some reason, but I still would like to mention:
I think it would make more sense to use static_assert, instead of making up our own macro for something that is in the standard already.
This is like making a mutex from scratch, instead of using pthread_mutex, cuz there is no pthread_mutex on Windows. We already depend on a lot of extensions outside of ISO, language extensions and POSIX specific, Linux specific features. I don't see how this would be different.